diff --git a/v1/yang/aft/openconfig-aft-common.yang b/v1/yang/aft/openconfig-aft-common.yang
index 65faad7..c9f4829 100644
--- a/v1/yang/aft/openconfig-aft-common.yang
+++ b/v1/yang/aft/openconfig-aft-common.yang
@@ -292,6 +292,18 @@ submodule openconfig-aft-common {
         specified header type.";
     }
 
+    leaf decapsulate-header {
+      type oc-aftt:encapsulation-header-type;
+      description
+        "When forwarding a packet to the specified next-hop, the local
+         system performs a decapsulation of the packet - removing the
+         specified header type. In the case that no next-hop is
+         specified, the packet header is removed, and a subsequent
+         forwarding lookup is performed on the packet encapsulated
+         within the header, matched within the relevant AFT within the
+         specified network-instance.";
+    }
+
     uses aft-common-install-protocol;
   }
 
